### PR TITLE
[CORE][FEATURE] Add Manager And Activities To Share Commits

### PR DESCRIPTION
**Pull request by chrismotian**
In order to be able to create PR's that based on the facade, I moved the changes from #364 to this branch instead of a branch in a fork. 
The reason why we wait to approve #364 : once all PR's with dependency changes are reviewed we approve all at once. The goal is to reduce the count of new builds the developers have to create.

edit: 
This PR is based on #428 to simplify reviews. 
This PR introduces the GitManager that handles activities that are necessary to share commits. The expected scenario is described in #324 . The activities to achieve that feature are introduced as well.

### CODE REVIEW

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/ISarosSession.java [hunk 1]
```diff
@@ -386,6 +388,12 @@ public void addSharedResources(
    */
   public void changeColor(int colorID);
 
+  public void gitSendCommitRequest();
```
Comment (srossbach) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/ISarosSession.java]: The ISarosSession interface is already in a bad state. There is no reason to pollute it further.

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/activities/GitSendBundleActivity.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package de.fu_berlin.inf.dpp.activities;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamAsAttribute;
+import de.fu_berlin.inf.dpp.session.User;
+import java.io.File;
+
+@XStreamAlias("GitSendBundleActivity")
+public class GitSendBundleActivity extends AbstractActivity {
```
[the remark this answers is not in the document]
Comment (chrismotian) [reply de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/activities/GitSendBundleActivity.java]: What is the problem?

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/ISarosSession.java (hunk 1 above)
Comment (chrismotian) [reply de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/ISarosSession.java]: done

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/git/JGitFacade.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+    Git git;
+    try {
+      git = Git.open(workDir);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("workDir can't be resolved", e);
+    }
+
+    try {
+      RemoteRemoveCommand rrc = git.remoteRemove();
```
[the remark this answers is not in the document]
Comment (chrismotian) [reply de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/git/JGitFacade.java]: done

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/git/JGitFacade.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+   * @throws Exception Error while adding the bundle as a remote or during fetch
+   */
+  public static void fetchFromBundle(File workDir, File bundleFile)
+      throws IllegalArgumentException, Exception {
+
+    Git git;
+    try {
+      git = Git.open(workDir);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("workDir can't be resolved", e);
```
[the remark this answers is not in the document]
Comment (chrismotian) [reply de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/git/JGitFacade.java]: done

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/git/JGitFacade.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+  /**
+   * Fetching from a bundle file to an Git repo
+   *
+   * @param workDir The directory that contains the .git directory
+   * @param bundleFile
+   * @throws IllegalArgumentException workDir or bundleFile can't be resolved
+   * @throws Exception Error while adding the bundle as a remote or during fetch
+   */
+  public static void fetchFromBundle(File workDir, File bundleFile)
+      throws IllegalArgumentException, Exception {
```
[the remark this answers is not in the document]
Comment (chrismotian) [reply de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/git/JGitFacade.java]: done

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/git/JGitFacade.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    // Step 4
+    File bundle;
+    try {
+      bundle = File.createTempFile("file", ".bundle");
+      OutputStream fos = new FileOutputStream(bundle);
+
+      NullProgressMonitor monitor = NullProgressMonitor.INSTANCE;
+
+      bundlewriter.writeBundle(monitor, fos);
+    } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (chrismotian) [reply de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/git/JGitFacade.java]: done

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/git/GitManager.java [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  private final ISarosSession session;
+
+  private File workDir = null;
+
+  public GitManager(ISarosSession session) {
+    this.session = session;
+  }
+
+  @Override
+  public synchronized void start() {
```
[the remark this answers is not in the document]
Comment (chrismotian) [reply de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/git/GitManager.java]: done

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/activities/GitSendBundleActivity.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrismotian) [reply de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/activities/GitSendBundleActivity.java]: done

**Pull request closed without merging**